Upstream-Status: Backport

Signed-off-by: Rob Woolley <rob.woolley@windriver.com>

From 270aaa87647ff1d6a1a7fdd0937b14ba106c8776 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Michael=20G=C3=B6rner?= <me@v4hn.de>
Date: Wed, 23 Apr 2025 21:27:09 +0200
Subject: [PATCH] [ROS-O] do not forward-declare Ogre types (#84)

* do not forward-declare Ogre types

In current Ogre these are actually typedef's and compilers
complain about an incongruent forward-declaration.

Also, there is no reason not to include the designated header here.

* replace obsolete use of boost _1
---
 interactive_marker_tutorials/src/pong.cpp      | 4 ++--
 interactive_marker_tutorials/src/selection.cpp | 2 +-
 rviz_plugin_tutorials/src/imu_visual.h         | 7 +------
 rviz_plugin_tutorials/src/plant_flag_tool.h    | 7 +------
 4 files changed, 5 insertions(+), 15 deletions(-)

diff --git a/rviz_plugin_tutorials/src/imu_visual.h b/rviz_plugin_tutorials/src/imu_visual.h
index 1334e180..020edc92 100644
--- a/src/imu_visual.h
+++ b/src/imu_visual.h
@@ -31,12 +31,7 @@
 #define IMU_VISUAL_H
 
 #include <sensor_msgs/Imu.h>
-
-namespace Ogre
-{
-class Vector3;
-class Quaternion;
-}
+#include <OgrePrerequisites.h>
 
 namespace rviz
 {
diff --git a/rviz_plugin_tutorials/src/plant_flag_tool.h b/rviz_plugin_tutorials/src/plant_flag_tool.h
index 21451a8e..afc41a03 100644
--- a/src/plant_flag_tool.h
+++ b/src/plant_flag_tool.h
@@ -30,12 +30,7 @@
 #define PLANT_FLAG_TOOL_H
 
 #include <rviz/tool.h>
-
-namespace Ogre
-{
-class SceneNode;
-class Vector3;
-}
+#include <OgrePrerequisites.h>
 
 namespace rviz
 {
